contrib/opencensus-ext-fastapi/opencensus/ext/fastapi/fastapi_middleware.py

```diff
@@ -0,0 +1,168 @@
+
+    :type app: :class: `~fastapi.FastAPI`
+    :param app: A fastapi application.
+
+    :type excludelist_paths: list
+    :param excludelist_paths: Paths that do not trace.
+
+    :type excludelist_hostnames: list
+    :param excludelist_hostnames: Hostnames that do not trace.
+
+    :type sampler: :class:`~opencensus.trace.samplers.base.Sampler`
+    :param sampler: A sampler. It should extend from the base
+                    :class:`.Sampler` type and implement
+                    :meth:`.Sampler.should_sample`. Defaults to
+                    :class:`.ProbabilitySampler`. Other options include
+                    :class:`.AlwaysOnSampler` and :class:`.AlwaysOffSampler`.
+
+    :type exporter: :class:`~opencensus.trace.base_exporter.exporter`
+    :param exporter: An exporter. Default to
+                     :class:`.PrintExporter`. The rest options are
+                     :class:`.FileExporter`, :class:`.LoggingExporter` and
+                     trace exporter extensions.
+
+    :type propagator: :class: 'object'
+    :param propagator: A propagator. Default to
+                       :class:`.TraceContextPropagator`. The rest options
+                       are :class:`.BinaryFormatPropagator`,
+                       :class:`.GoogleCloudFormatPropagator` and
+                       :class:`.TextFormatPropagator`.
+    """
+
+    def __init__(
+        self,
+        app: ASGIApp,
+        excludelist_paths=None,
+        excludelist_hostnames=None,
+        sampler=None,
+        exporter=None,
+        propagator=None,
+    ) -> None:
+        super().__init__(app)
+        self.excludelist_paths = excludelist_paths
+        self.excludelist_hostnames = excludelist_hostnames
+        self.sampler = sampler or samplers.AlwaysOnSampler()
+        self.exporter = exporter or print_exporter.PrintExporter()
+        self.propagator = propagator or trace_context_http_header_format.TraceContextPropagator()
+
+        # pylint: disable=protected-access
+        integrations.add_integration(integrations._Integrations.FASTAPI)
+
+    def _prepare_tracer(self, request: Request) -> tracer_module.Tracer:
+        span_context = self.propagator.from_headers(request.headers)
+        tracer = tracer_module.Tracer(
+            span_context=span_context,
+            sampler=self.sampler,
+            exporter=self.exporter,
+            propagator=self.propagator,
+        )
+        return tracer
+
+    def _before_request(self, span: Union[Span, BlankSpan], request: Request):
+        span.span_kind = span_module.SpanKind.SERVER
+        span.name = "[{}]{}".format(request.method, request.url)
+        span.add_attribute(HTTP_HOST, request.url.hostname)
+        span.add_attribute(HTTP_METHOD, request.method)
+        span.add_attribute(HTTP_PATH, request.url.path)
+        span.add_attribute(HTTP_URL, str(request.url))
+        span.add_attribute(HTTP_ROUTE, request.url.path)
+        execution_context.set_opencensus_attr(
```

Comment:
Curious what this is for?

Reply:
@lzchen Thanks for pointing this out. I added it because it was present in [the original implementation](https://github.com/census-instrumentation/opencensus-python/issues/1020#issue-852473135) and also in the [flask](https://github.com/census-instrumentation/opencensus-python/blob/b85e4767edcd6524515d263853247335f9ee69f4/contrib/opencensus-ext-flask/opencensus/ext/flask/flask_middleware.py#L173-L176) and [django](https://github.com/census-instrumentation/opencensus-python/blob/fd064f438c5e490d25b004ee2545be55d2e28679/contrib/opencensus-ext-django/opencensus/ext/django/middleware.py#L194-L196) implementations. To be honest, I don't really understand the point here, although it seems odd that hostnames is there but paths is not. Should I remove it?

Reply:
I'm assuming the original intention was to the exclude list as a global list of hostnames to not track (across integrations). Doesn't seem like that feature is implemented though. The way you have it now is fine :)